pkg/controllers/termination/v1alpha1/terminate.go

```diff
@@ -139,13 +123,15 @@ func (t *Terminator) deleteNodes(ctx context.Context, nodes []*v1.Node, provisio
 	return nil
 }
 
-// getNodes returns a list of nodes with the provisioner's labels and given labels
-func (t *Terminator) getNodes(ctx context.Context, provisioner *v1alpha1.Provisioner, additionalLabels map[string]string) ([]*v1.Node, error) {
+// getLabeledNodes returns a list of nodes with the provisioner's labels and given labels
+func (t *Terminator) getLabeledNodes(ctx context.Context, phaseLabel string) ([]*v1.Node, error) {
```

Comment:
When we implement watches, we will reconcile nodes themselves. This means we won't need to query nodes, and this function will be deleted as part of that change.  I'll add the enum for this.

Reply:
We define the phases in `provisioner.go` and use it to patch label maps of nodes. This would ultimately lead to a decent amount of string casting, which I think would be a redundant use of enum here. WDYT?

Reply:
What if the strings were themselves already the right type where they were defined :). I'm fine just using raw types though, probably cleaner overall -- I haven't played with it myself.

Reply:
It's a little tricky when we use the labels in `utilization.go` as the label maps are map[string]string, and not passed in as a parameter. I like the idea, and think that either way would be the same net amount of code. I'll keep it the way as is, as we are changing it in the next PR.